astroid/brain/brain_builtin_inference.py

```diff
@@ -892,6 +893,21 @@ def _build_dict_with_elements(elements):
     return _build_dict_with_elements([])
 
 
+def _infer_copy_method(
+    node: nodes.Call, context: Optional[InferenceContext] = None
+) -> Iterator[Union[nodes.Dict, nodes.List, nodes.Set, objects.FrozenSet]]:
```

Comment:
```suggestion
) -> Iterator[nodes.NodeNG]:
```
The result is only use the check the function itself, nothing else. So it's not necessary to constrict it too much.